module_2/src/com/epam/arrraysofarrays/MatrixModifier.java

```diff
@@ -0,0 +1,21 @@
+package com.epam.arrraysofarrays;
+
+public class MatrixModifier {
+
+    // task_08
+    public static boolean replaceColumns(int[][] matrix, int firstColumn, int secondColumn) {
+        boolean isModified = false;
+        if (firstColumn > 0 && firstColumn <= matrix[0].length && secondColumn > 0 && secondColumn <= matrix[0].length
+                && firstColumn != secondColumn) {
+            int temp;
+            for (int i = 0; i < matrix.length; i++) {
+                temp = matrix[i][firstColumn - 1];
+                matrix[i][firstColumn - 1] = matrix[i][secondColumn - 1];
+                matrix[i][secondColumn - 1] = temp;
+            }
```

Comment:
"firstColumn - 1" and "secondColumn - 1" are calculated every time in the for-cycle, so you can improve it like
```suggestion
            final int firstRealColumn = firstColumn - 1;
            final int secondRealColumn = secondColumn - 1;
            for (int i = 0; i < matrix.length; i++) {
                temp = matrix[i][firstRealColumn];
                matrix[i][firstRealColumn] = matrix[i][secondRealColumn];
                matrix[i][secondRealColumn] = temp;
            }
```

Reply:
I'll bring it into line with the java numbering rules too (without decrementing of parameters)